[0.9.0] SELECT * without database causes panic

Not entirely sure what caused this error as I can't reproduce but here is the entire log from start up to crash. Basically it starts up and I write some data from collectd and then perform some queries. Then it crashes.  I haven't been able to reproduce so hopefully the log is enough to find the panic. Let me know if I can provide more information. 

Version 0.9.0
Host CentOS 7 

```
GOMAXPROCS set to 24
2015/06/11 10:20:01 [INFO] raft: Node at 127.0.0.1:8088 [Follower] entering Follower state
2015/06/11 10:20:01 read local node id: 1
2015/06/11 10:20:03 [WARN] raft: Heartbeat timeout reached, starting election
2015/06/11 10:20:03 [INFO] raft: Node at 127.0.0.1:8088 [Candidate] entering Candidate state
2015/06/11 10:20:03 [DEBUG] raft: Votes needed: 1
2015/06/11 10:20:03 [DEBUG] raft: Vote granted. Tally: 1
2015/06/11 10:20:03 [INFO] raft: Election won. Tally: 1
2015/06/11 10:20:03 [INFO] raft: Node at 127.0.0.1:8088 [Leader] entering Leader state
2015/06/11 10:20:03 [INFO] raft: Disabling EnableSingleNode (bootstrap)
2015/06/11 10:20:03 [DEBUG] raft: Node 127.0.0.1:8088 updated peer set (2): [127.0.0.1:8088]
2015/06/11 10:20:03 [DEBUG] raft: Node 127.0.0.1:8088 updated peer set (2): [127.0.0.1:8088]
2015/06/11 10:20:03 [DEBUG] raft: Node 127.0.0.1:8088 updated peer set (2): [127.0.0.1:8088]
[httpd] 2015/06/11 10:20:03 listening on HTTP: [::]:8086
[collectd] 2015/06/11 10:20:03 collectd UDP started
2015/06/11 10:20:03 Sending anonymous usage statistics to m.influxdb.com
[http] 2015/06/11 10:20:45 10.1.50.42 - - [11/Jun/2015:10:20:45 -0600] GET /query?u=*&p=*&q=show%20measurements HTTP/1.1 200 40 http://influxdb:8083/ Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 d00501a6-1055-11e5-8001-000000000000 883.763µs
[http] 2015/06/11 10:20:57 10.1.50.42 - - [11/Jun/2015:10:20:57 -0600] GET /query?u=*&p=*&q=show%20measurements HTTP/1.1 200 40 http://influxdb:8083/ Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 d719ea5b-1055-11e5-8002-000000000000 4.236443ms
[http] 2015/06/11 10:21:10 10.1.50.42 - - [11/Jun/2015:10:21:10 -0600] GET /query?u=*&p=*&q=show%20measurements HTTP/1.1 200 40 http://influxdb:8083/ Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 dece869e-1055-11e5-8003-000000000000 426.646µs
[http] 2015/06/11 10:21:31 10.1.50.42 - - [11/Jun/2015:10:21:31 -0600] GET /query?u=*&p=*&q=show%20measurements HTTP/1.1 200 40 http://influxdb:8083/ Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 ebbadc94-1055-11e5-8004-000000000000 439.758µs
[http] 2015/06/11 10:22:16 10.1.50.42 - - [11/Jun/2015:10:22:16 -0600] GET /query?u=*&p=*&q=SHOW%20DATABASES HTTP/1.1 200 95 http://influxdb:8083/ Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 06416026-1056-11e5-8005-000000000000 587.513µs
[http] 2015/06/11 10:22:19 10.1.50.42 - - [11/Jun/2015:10:22:19 -0600] GET /query?u=*&p=*&q=SHOW%20DATABASES HTTP/1.1 200 95 http://influxdb:8083/ Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 086e545d-1056-11e5-8006-000000000000 457.87µs
[http] 2015/06/11 10:22:52 10.1.50.42 - - [11/Jun/2015:10:22:52 -0600] GET /query?u=*&p=*&q=show%20measurements&db=collectd HTTP/1.1 200 192 http://influxdb:8083/ Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 1c0b22e6-1056-11e5-8007-000000000000 609.671µs
[http] 2015/06/11 10:23:23 10.1.50.42 - - [11/Jun/2015:10:23:22 -0600] GET /query?u=*&p=*&q=show%20series&db=collectd HTTP/1.1 200 104761 http://influxdb:8083/ Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 2dfb7d8b-1056-11e5-8008-000000000000 178.950067ms
[http] 2015/06/11 10:26:21 127.0.0.1 - - [11/Jun/2015:10:26:21 -0600] GET /ping HTTP/1.1 204 0 - InfluxDBShell/0.9.0 983e9d9a-1056-11e5-8009-000000000000 85.226µs
[http] 2015/06/11 10:26:28 127.0.0.1 - - [11/Jun/2015:10:26:28 -0600] GET /query?db=&q=show+databases%3B HTTP/1.1 200 95 - InfluxDBShell/0.9.0 9c860250-1056-11e5-800a-000000000000 549.209µs
[http] 2015/06/11 10:26:45 127.0.0.1 - - [11/Jun/2015:10:26:45 -0600] GET /query?db=collectd%3B&q=show+measurements%3B HTTP/1.1 200 40 - InfluxDBShell/0.9.0 a6de3fe9-1056-11e5-800b-000000000000 446.35µs
[http] 2015/06/11 10:26:48 127.0.0.1 - - [11/Jun/2015:10:26:48 -0600] GET /query?db=collectd%3B&q=show+measurements%3B HTTP/1.1 200 40 - InfluxDBShell/0.9.0 a86b6dcd-1056-11e5-800c-000000000000 417.46µs
[http] 2015/06/11 10:26:54 10.1.50.42 - - [11/Jun/2015:10:26:54 -0600] GET /query?u=*&p=*&q=show%20measurements&db=collectd HTTP/1.1 200 192 http://influxdb:8083/ Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 ac00c4bb-1056-11e5-800d-000000000000 542.715µs
[http] 2015/06/11 10:26:59 127.0.0.1 - - [11/Jun/2015:10:26:59 -0600] GET /query?db=collectd%3B&q=show+measurements HTTP/1.1 200 40 - InfluxDBShell/0.9.0 aed37a81-1056-11e5-800e-000000000000 474.527µs
[http] 2015/06/11 10:27:00 127.0.0.1 - - [11/Jun/2015:10:27:00 -0600] GET /query?db=collectd%3B&q=%3B HTTP/1.1 200 26 - InfluxDBShell/0.9.0 af80788d-1056-11e5-800f-000000000000 369.568µs
[http] 2015/06/11 10:27:04 127.0.0.1 - - [11/Jun/2015:10:27:04 -0600] GET /query?db=collectd%3B&q=show+measurements HTTP/1.1 200 40 - InfluxDBShell/0.9.0 b2077507-1056-11e5-8010-000000000000 383.449µs
[http] 2015/06/11 10:27:07 10.1.50.42 - - [11/Jun/2015:10:27:07 -0600] GET /query?u=*&p=*&q=show%20measurements&db=collectd HTTP/1.1 200 192 http://influxdb:8083/ Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 b4085fa8-1056-11e5-8011-000000000000 743.682µs
[http] 2015/06/11 10:27:32 10.1.50.42 - - [11/Jun/2015:10:27:32 -0600] GET /query?u=*&p=*&q=select%20*%20from%20cpu_value%20where%20host%20%3D%20%27collector1%27&db=collectd HTTP/1.1 200 4994 http://influxdb:8083/ Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 c2dcdab0-1056-11e5-8012-000000000000 6.712154ms
[http] 2015/06/11 10:27:52 10.1.50.42 - - [11/Jun/2015:10:27:52 -0600] GET /query?u=*&p=*&q=select%20*%20from%20cpu_value%20where%20host%20%3D%20%27collector1%27%20group%20by%20*&db=collectd HTTP/1.1 200 5193 http://influxdb:8083/ Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 ceb2a93d-1056-11e5-8013-000000000000 6.588366ms
[http] 2015/06/11 10:28:40 127.0.0.1 - - [11/Jun/2015:10:28:40 -0600] GET /ping HTTP/1.1 204 0 - InfluxDBShell/0.9.0 eb435a5a-1056-11e5-8014-000000000000 78.18µs

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x66955f]

goroutine 336 [running]:
github.com/influxdb/influxdb/influxql.(*SelectStatement).RewriteDistinct(0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/ast.go:780 +0x37f
github.com/influxdb/influxdb/tsdb.(*QueryExecutor).rewriteSelectStatement(0xc208097840, 0x0, 0x7fa5744105d8, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:271 +0x18a
github.com/influxdb/influxdb/tsdb.(*QueryExecutor).executeSelectStatement(0xc208097840, 0x0, 0xc20911a000, 0xc21157965b, 0x0, 0xc20e311c80, 0x2710, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:219 +0x58
github.com/influxdb/influxdb/tsdb.func·004()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:157 +0x727
created by github.com/influxdb/influxdb/tsdb.(*QueryExecutor).ExecuteQuery
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:211 +0x22b

goroutine 1 [chan receive (nil chan), 8 minutes]:
main.(*Main).Run(0xc208019760, 0xc20800a010, 0x4, 0x4, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/main.go:81 +0x450
main.main()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/main.go:39 +0x1d0

goroutine 6 [syscall, 8 minutes]:
os/signal.loop()
	/root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:27 +0x35

goroutine 17 [syscall, 8 minutes, locked to thread]:
runtime.goexit()
	/root/.gvm/gos/go1.4.2/src/runtime/asm_amd64.s:2232 +0x1

goroutine 8 [IO wait, 8 minutes]:
net.(*pollDesc).Wait(0xc20800c530, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800c530, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc20800c4d0, 0x0, 0x7fa577da3cc8, 0xc208102000)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc2080368f0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc2080368f0, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:244 +0x4c
github.com/influxdb/influxdb/tcp.(*Mux).Serve(0xc2080c1920, 0x7fa577daa250, 0xc2080368f0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:45 +0x57
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:274 +0x762

goroutine 9 [chan receive, 8 minutes]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc2080368f8, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:117 +0x80
github.com/influxdb/influxdb/meta.(*raftLayer).Accept(0xc2080c1950, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/meta/store.go:1636 +0x64
github.com/hashicorp/raft.(*NetworkTransport).listen(0xc20800c5b0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/net_transport.go:346 +0x50
created by github.com/hashicorp/raft.NewNetworkTransport
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/net_transport.go:138 +0x2c4

goroutine 10 [select]:
github.com/hashicorp/raft.(*Raft).leaderLoop(0xc2080be1c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:807 +0xc7c
github.com/hashicorp/raft.(*Raft).runLeader(0xc2080be1c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:779 +0x74e
github.com/hashicorp/raft.(*Raft).run(0xc2080be1c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:576 +0xc9
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.run)·fm()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:241 +0x27
github.com/hashicorp/raft.func·011()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 11 [select, 8 minutes]:
github.com/hashicorp/raft.(*Raft).runFSM(0xc2080be1c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:490 +0xf74
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.runFSM)·fm()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:242 +0x27
github.com/hashicorp/raft.func·011()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 12 [select]:
github.com/hashicorp/raft.(*Raft).runSnapshots(0xc2080be1c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:1594 +0x466
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.runSnapshots)·fm()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:243 +0x27
github.com/hashicorp/raft.func·011()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 13 [chan receive, 8 minutes]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc208036900, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:117 +0x80
github.com/influxdb/influxdb/meta.(*Store).serveExecListener(0xc208078280)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/meta/store.go:436 +0x8f
created by github.com/influxdb/influxdb/meta.(*Store).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/meta/store.go:183 +0x1c2

goroutine 14 [select, 8 minutes]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).monitorErrorChan(0xc2080a40c0, 0xc20804c4e0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:390 +0x14a
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:280 +0x8d3

goroutine 15 [select]:
github.com/influxdb/influxdb/services/hh.(*Service).retryWrites(0xc208057220)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/hh/service.go:98 +0x2e7
created by github.com/influxdb/influxdb/services/hh.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/hh/service.go:62 +0xef

goroutine 16 [select, 8 minutes]:
github.com/influxdb/influxdb/services/hh.(*Service).expireWrites(0xc208057220)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/hh/service.go:116 +0x2a4
created by github.com/influxdb/influxdb/services/hh.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/hh/service.go:63 +0x109

goroutine 34 [chan receive, 8 minutes]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc208036908, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:117 +0x80
github.com/influxdb/influxdb/cluster.(*Service).serve(0xc20800c380)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cluster/service.go:79 +0x102
created by github.com/influxdb/influxdb/cluster.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cluster/service.go:56 +0x67

goroutine 35 [select, 8 minutes]:
github.com/influxdb/influxdb/services/precreator.(*Service).runPrecreation(0xc20800adc0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/precreator/service.go:72 +0x407
created by github.com/influxdb/influxdb/services/precreator.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/precreator/service.go:50 +0xcb

goroutine 36 [chan receive, 8 minutes]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc208036910, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:117 +0x80
github.com/influxdb/influxdb/services/snapshotter.(*Service).serve(0xc20800aeb0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/snapshotter/service.go:68 +0xad
created by github.com/influxdb/influxdb/services/snapshotter.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/snapshotter/service.go:46 +0x62

goroutine 37 [IO wait, 7 minutes]:
net.(*pollDesc).Wait(0xc20800db10, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800db10, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc20800dab0, 0x0, 0x7fa577da3cc8, 0xc20a2c0e20)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc208036a30, 0xc20818bd90, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc208036a30, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc20804c0c0, 0x7fa577daa250, 0xc208036a30, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1728 +0x92
net/http.Serve(0x7fa577daa250, 0xc208036a30, 0x7fa577dabcf8, 0xc208028920, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1606 +0xa1
github.com/influxdb/influxdb/services/admin.(*Service).serve(0xc2080c1710)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/admin/service.go:71 +0x147
created by github.com/influxdb/influxdb/services/admin.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/admin/service.go:39 +0xd9

goroutine 38 [select]:
github.com/influxdb/influxdb/services/continuous_querier.(*Service).backgroundLoop(0xc20800c3f0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:151 +0x1fc
created by github.com/influxdb/influxdb/services/continuous_querier.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:87 +0x1ed

goroutine 39 [IO wait]:
net.(*pollDesc).Wait(0xc20800db80, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800db80, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc20800db20, 0x0, 0x7fa577da3cc8, 0xc20a005e30)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc208036a38, 0xc20802edc8, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc208036a38, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc208120180, 0x7fa577daa250, 0xc208036a38, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1728 +0x92
net/http.Serve(0x7fa577daa250, 0xc208036a38, 0x7fa577dabc58, 0xc20800c460, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1606 +0xa1
github.com/influxdb/influxdb/services/httpd.(*Service).serve(0xc2080978c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/service.go:83 +0x79
created by github.com/influxdb/influxdb/services/httpd.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/service.go:51 +0x283

goroutine 50 [select]:
github.com/influxdb/influxdb/tsdb.func·002()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/batcher.go:69 +0x512
created by github.com/influxdb/influxdb/tsdb.(*PointBatcher).Start
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/batcher.go:102 +0x29a

goroutine 51 [IO wait]:
net.(*pollDesc).Wait(0xc208148140, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208148140, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).readFrom(0xc2081480e0, 0xc208102000, 0x5ac, 0x5ac, 0x0, 0x0, 0x0, 0x7fa577da3cc8, 0xc2094be078)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:269 +0x4a1
net.(*UDPConn).ReadFromUDP(0xc2080366c8, 0xc208102000, 0x5ac, 0x5ac, 0x529, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/udpsock_posix.go:67 +0x124
github.com/influxdb/influxdb/services/collectd.(*Service).serve(0xc2080b4a80)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/collectd/service.go:167 +0x113
created by github.com/influxdb/influxdb/services/collectd.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/collectd/service.go:102 +0xa81

goroutine 52 [select]:
github.com/influxdb/influxdb/cluster.(*PointsWriter).WritePoints(0xc20800c310, 0xc209499340, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cluster/points_writer.go:210 +0x634
github.com/influxdb/influxdb/services/collectd.(*Service).writePoints(0xc2080b4a80)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/collectd/service.go:206 +0x28d
created by github.com/influxdb/influxdb/services/collectd.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/collectd/service.go:103 +0xa9e

goroutine 53 [select, 8 minutes]:
github.com/influxdb/influxdb/services/retention.(*Service).deleteShardGroups(0xc20804c960)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/retention/service.go:67 +0x433
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/retention/service.go:44 +0x67

goroutine 54 [select, 8 minutes]:
github.com/influxdb/influxdb/services/retention.(*Service).deleteShards(0xc20804c960)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/retention/service.go:101 +0x821
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/retention/service.go:45 +0x81

goroutine 55 [chan receive, 8 minutes]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).startServerReporting(0xc2080a40c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:341 +0x5e
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:303 +0xd78

goroutine 56 [select, 8 minutes]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Command).monitorServerErrors(0xc20804c360)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:118 +0x2da
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Command).Run
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:102 +0xa1a

goroutine 46 [IO wait]:
net.(*pollDesc).Wait(0xc209727e90, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc209727e90, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc209727e30, 0xc209b44000, 0x1000, 0x1000, 0x0, 0x7fa577da3cc8, 0xc209f4f390)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc2081252a8, 0xc209b44000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/net.go:121 +0xdc
net/http.(*liveSwitchReader).Read(0xc2080560e8, 0xc209b44000, 0x1000, 0x1000, 0xc20800af50, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:214 +0xab
io.(*LimitedReader).Read(0xc20a01a8e0, 0xc209b44000, 0x1000, 0x1000, 0xc2091ef940, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/io/io.go:408 +0xce
bufio.(*Reader).fill(0xc208ebc420)
	/root/.gvm/gos/go1.4.2/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadSlice(0xc208ebc420, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/bufio/bufio.go:295 +0x257
bufio.(*Reader).ReadLine(0xc208ebc420, 0x0, 0x0, 0x0, 0xc207e90c00, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/bufio/bufio.go:324 +0x62
net/textproto.(*Reader).readLineSlice(0xc2096f3440, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/textproto/reader.go:55 +0x9e
net/textproto.(*Reader).ReadLine(0xc2096f3440, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/textproto/reader.go:36 +0x4f
net/http.ReadRequest(0xc208ebc420, 0xc20a9afad0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/request.go:598 +0xcb
net/http.(*conn).readRequest(0xc2080560a0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:586 +0x26f
net/http.(*conn).serve(0xc2080560a0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1162 +0x69e
created by net/http.(*Server).Serve
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1751 +0x35e

goroutine 335 [chan receive]:
github.com/influxdb/influxdb/services/httpd.(*Handler).serveQuery(0xc20800c460, 0x7fa574577ba0, 0xc2087246e0, 0xc208147110, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:241 +0x7c4
github.com/influxdb/influxdb/services/httpd.*Handler.(github.com/influxdb/influxdb/services/httpd.serveQuery)·fm(0x7fa574577ba0, 0xc2087246e0, 0xc208147110, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:90 +0x4f
github.com/influxdb/influxdb/services/httpd.func·001(0x7fa574577ba0, 0xc2087246e0, 0xc208147110)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:601 +0x78
net/http.HandlerFunc.ServeHTTP(0xc2080f0e60, 0x7fa574577ba0, 0xc2087246e0, 0xc208147110)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·002(0x7fa574577ac8, 0xc208724680, 0xc208147110)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:659 +0x288
net/http.HandlerFunc.ServeHTTP(0xc2080f24a0, 0x7fa574577ac8, 0xc208724680, 0xc208147110)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·003(0x7fa574577ac8, 0xc208724680, 0xc208147110)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:668 +0xb0
net/http.HandlerFunc.ServeHTTP(0xc2080f0e80, 0x7fa574577ac8, 0xc208724680, 0xc208147110)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·004(0x7fa574577ac8, 0xc208724680, 0xc208147110)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:701 +0x31a
net/http.HandlerFunc.ServeHTTP(0xc2080f24e0, 0x7fa574577ac8, 0xc208724680, 0xc208147110)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·005(0x7fa574577ac8, 0xc208724680, 0xc208147110)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:711 +0x179
net/http.HandlerFunc.ServeHTTP(0xc2080f2500, 0x7fa574577ac8, 0xc208724680, 0xc208147110)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·006(0x7fa574577ac8, 0xc208724640, 0xc208147110)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:719 +0x123
net/http.HandlerFunc.ServeHTTP(0xc2080f0ea0, 0x7fa574577ac8, 0xc208724640, 0xc208147110)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·007(0x7fa574577a90, 0xc208056320, 0xc208147110)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:729 +0x12c
net/http.HandlerFunc.ServeHTTP(0xc2080f0ec0, 0x7fa574577a90, 0xc208056320, 0xc208147110)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/bmizerany/pat.(*PatternServeMux).ServeHTTP(0xc208036840, 0x7fa574577a90, 0xc208056320, 0xc208147110)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/bmizerany/pat/mux.go:109 +0x21c
github.com/influxdb/influxdb/services/httpd.(*Handler).ServeHTTP(0xc20800c460, 0x7fa574577a90, 0xc208056320, 0xc208147110)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:166 +0x256
net/http.serverHandler.ServeHTTP(0xc208120180, 0x7fa574577a90, 0xc208056320, 0xc208147110)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc208056140)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1751 +0x35e

goroutine 360 [runnable]:
github.com/boltdb/bolt._assert(0xc20a53a101, 0xb5c490, 0x1b, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/boltdb/bolt/db.go:780
github.com/boltdb/bolt.(*node).read(0xc20bd97500, 0x7fa51da64000)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/boltdb/bolt/node.go:178 +0x2f0
github.com/boltdb/bolt.(*Bucket).node(0xc208da7640, 0x1a64, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/boltdb/bolt/bucket.go:632 +0x1cb
github.com/boltdb/bolt.(*Cursor).node(0xc2085e8780, 0xc2108f8cf0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/boltdb/bolt/cursor.go:353 +0x154
github.com/boltdb/bolt.(*Bucket).Put(0xc208da7640, 0xc2108f8cf8, 0x8, 0x8, 0xc208723ed0, 0x9, 0xa, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/boltdb/bolt/bucket.go:298 +0x452
github.com/influxdb/influxdb/tsdb.func·009(0xc20a9afba0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/shard.go:187 +0x64a
github.com/boltdb/bolt.(*DB).Update(0xc208042820, 0xc20e8f5e68, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/boltdb/bolt/db.go:547 +0xde
github.com/influxdb/influxdb/tsdb.(*Shard).WritePoints(0xc2083da800, 0xc20b0e4000, 0x1388, 0x1800, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/shard.go:193 +0x523
github.com/influxdb/influxdb/tsdb.(*Store).WriteToShard(0xc208097800, 0x1, 0xc20b0e4000, 0x1388, 0x1800, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/store.go:260 +0xf7
github.com/influxdb/influxdb/cluster.func·002(0x1, 0x1, 0xc20b0e4000, 0x1388, 0x1800)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cluster/points_writer.go:241 +0xfc
created by github.com/influxdb/influxdb/cluster.(*PointsWriter).writeToShard
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cluster/points_writer.go:271 +0x30b

goroutine 359 [select]:
github.com/influxdb/influxdb/cluster.(*PointsWriter).writeToShard(0xc20800c310, 0xc20874a4a0, 0xc208052a40, 0x8, 0xc20814a2b8, 0x7, 0x0, 0xc20b0e4000, 0x1388, 0x1800, ...)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cluster/points_writer.go:278 +0x90e
github.com/influxdb/influxdb/cluster.func·001(0xc20874a4a0, 0xc208052a40, 0x8, 0xc20814a2b8, 0x7, 0xc20b0e4000, 0x1388, 0x1800)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cluster/points_writer.go:205 +0xad
created by github.com/influxdb/influxdb/cluster.(*PointsWriter).WritePoints
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cluster/points_writer.go:206 +0x447
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,7 @@
 - [#2928](https://github.com/influxdb/influxdb/pull/2928): Start work to set InfluxDB version in HTTP response headers. Thanks @neonstalwart.
 - [#2969](https://github.com/influxdb/influxdb/pull/2969): Actually set HTTP version in responses.
 - [#2993](https://github.com/influxdb/influxdb/pull/2993): Don't log each UDP batch.
+- [#2994](https://github.com/influxdb/influxdb/pull/2994): Don't panic during wilcard expanasion if no default database specified.
 
 ## v0.9.0 [2015-06-11]
 

--- a/cmd/influxd/run/server_test.go
+++ b/cmd/influxd/run/server_test.go
@@ -847,7 +847,7 @@ func TestServer_Query_Common(t *testing.T) {
 		&Query{
 			name:    "selecting a from a non-existent database should error",
 			command: `SELECT value FROM db1.rp0.cpu`,
-			exp:     `{"results":[{"error":"database not found"}]}`,
+			exp:     `{"results":[{"error":"database not found: db1"}]}`,
 		},
 		&Query{
 			name:    "selecting a from a non-existent retention policy should error",
@@ -869,6 +869,16 @@ func TestServer_Query_Common(t *testing.T) {
 			name:    "selecting a field that doesn't exist should error",
 			command: `SELECT idontexist FROM db0.rp0.cpu`,
 			exp:     `{"results":[{"error":"unknown field or tag name in select clause: idontexist"}]}`,
+		},
+		&Query{
+			name:    "selecting wildcard without specifying a database should error",
+			command: `SELECT * FROM cpu`,
+			exp:     `{"results":[{"error":"database name required"}]}`,
+		},
+		&Query{
+			name:    "selecting explicit field without specifying a database should error",
+			command: `SELECT value FROM cpu`,
+			exp:     `{"results":[{"error":"database name required"}]}`,
 		},
 	}...)
 

--- a/tsdb/query_executor.go
+++ b/tsdb/query_executor.go
@@ -154,7 +154,7 @@ func (q *QueryExecutor) ExecuteQuery(query *influxql.Query, database string, chu
 			var res *influxql.Result
 			switch stmt := stmt.(type) {
 			case *influxql.SelectStatement:
-				if err := q.executeSelectStatement(i, stmt, database, results, chunkSize); err != nil {
+				if err := q.executeSelectStatement(i, stmt, results, chunkSize); err != nil {
 					results <- &influxql.Result{Err: err}
 					break
 				}
@@ -214,7 +214,7 @@ func (q *QueryExecutor) ExecuteQuery(query *influxql.Query, database string, chu
 }
 
 // executeSelectStatement plans and executes a select statement against a database.
-func (q *QueryExecutor) executeSelectStatement(statementID int, stmt *influxql.SelectStatement, database string, results chan *influxql.Result, chunkSize int) error {
+func (q *QueryExecutor) executeSelectStatement(statementID int, stmt *influxql.SelectStatement, results chan *influxql.Result, chunkSize int) error {
 	// Perform any necessary query re-writing.
 	stmt, err := q.rewriteSelectStatement(stmt)
 	if err != nil {

--- a/tsdb/query_executor.go
+++ b/tsdb/query_executor.go
@@ -134,8 +134,9 @@ func (q *QueryExecutor) ExecuteQuery(query *influxql.Query, database string, chu
 		var i int
 		var stmt influxql.Statement
 		for i, stmt = range query.Statements {
-			// If a default database wasn't passed in by the caller,
-			// try to get it from the statement.
+			// If a default database wasn't passed in by the caller, check the statement.
+			// Some types of statements have an associated default database, even if it
+			// is not explicitly included.
 			defaultDB := database
 			if defaultDB == "" {
 				if s, ok := stmt.(influxql.HasDefaultDatabase); ok {
@@ -143,12 +144,10 @@ func (q *QueryExecutor) ExecuteQuery(query *influxql.Query, database string, chu
 				}
 			}
 
-			// If we have a default database, normalize the statement with it.
-			if defaultDB != "" {
-				if err := q.normalizeStatement(stmt, defaultDB); err != nil {
-					results <- &influxql.Result{Err: err}
-					break
-				}
+			// Normalize each statement.
+			if err := q.normalizeStatement(stmt, defaultDB); err != nil {
+				results <- &influxql.Result{Err: err}
+				break
 			}
 
 			var res *influxql.Result
@@ -905,17 +904,21 @@ func (q *QueryExecutor) normalizeStatement(stmt influxql.Statement, defaultDatab
 	return
 }
 
-// normalizeMeasurement inserts the default database or policy into all measurement names.
+// normalizeMeasurement inserts the default database or policy into all measurement names,
+// if required.
 func (q *QueryExecutor) normalizeMeasurement(m *influxql.Measurement, defaultDatabase string) error {
-	if defaultDatabase == "" {
-		return errors.New("no default database specified")
-	}
 	if m.Name == "" && m.Regex == nil {
 		return errors.New("invalid measurement")
 	}
 
+	// Measurement does not have an explicit database? Insert default.
 	if m.Database == "" {
 		m.Database = defaultDatabase
+	}
+
+	// The database must now be specified by this point.
+	if m.Database == "" {
+		return errors.New("database name required")
 	}
 
 	// Find database.

